src/main/java/org/gridsuite/shortcircuit/server/dto/ResourceFilter.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.shortcircuit.server.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.List;
+
+/**
+ * An object that can be used to filter data with the JPA Criteria API (via Spring Specification)
+ * @param dataType the type of data we want to filter (text, number)
+ * @param type the type of filter (contains, startsWith...)
+ * @param value the value of the filter
+ * @param field the field on which the filter will be applied
+ */
+public record ResourceFilter(DataType dataType, Type type, String value, String field) {
```

Comment:
Could you add @author comment please on some files you just created or edited ?

Reply:
Ok done